src/sessions/live-monitoring-session-handler.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  IPendingSession,
+  IExtendedMediaSession,
+  IAcceptSessionRequest,
+  IUpdateOutgoingMedia,
+} from '../types/interfaces';
+import BaseSessionHandler from './base-session-handler';
+import { SessionTypes, SdkErrorTypes } from '../types/enums';
+import {createAndEmitSdkError, isLiveScreenMonitorJid, requestApi} from '../utils';
+import {Constants} from "stanza";
+
+export class LiveMonitoringSessionHandler extends BaseSessionHandler {
+  sessionType = SessionTypes.liveScreenMonitoring;
+
+  shouldHandleSessionByJid(jid: string): boolean {
+    return isLiveScreenMonitorJid(jid);
+  }
+
+  handleConversationUpdate(): void {
+    /* no-op */
+    return;
+  }
+
+  async handlePropose (pendingSession: IPendingSession): Promise<void> {
+    if (this.sdk._config.autoAcceptPendingLiveScreenMonitoringRequests) {
+      return this.proceedWithSession(pendingSession);
+    }
+
+    // if not auto accepting sessions, emit this event for the consumer to accept
+    await super.handlePropose(pendingSession);
+  }
+
+  async acceptSession(session: IExtendedMediaSession, params: IAcceptSessionRequest): Promise<any> {
+    if (!params.mediaStream) {
+      throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.session, `Cannot accept screen recording session without providing a media stream`, { conversationId: session.conversationId, sessionType: this.sessionType });
+    }
+
+    if (!params.screenRecordingMetadatas?.length) {
```

Comment:
Just to confirm: we're re-using the `screenRecordingMetadata` field rather than having a separate one for live screen monitoring?

Related: do we actually use the metadata for anything? I noticed that for screen recording, we update it when the session connects, but it looks like we don't need to do that for live screen monitoring?